lib/embedded_ansible.rb

```diff
@@ -43,9 +43,41 @@ def api_connection_raw(host, port)
     )
   end
 
+  def find_or_create_secret_key
+    miq_database.ansible_secret_key || miq_database.ansible_secret_key = SecureRandom.hex(16)
+  end
+
+  def find_or_create_admin_authentication
+    miq_database.ansible_admin_authentication || miq_database.set_ansible_admin_authentication(:password => generate_password)
```

Comment:
Can these also be moved to `miq_database.ansible_admin_authentication=` then collapsed to `||=` like https://github.com/ManageIQ/manageiq/pull/16205/commits/52bab34dee3f230c12eac594aae35ff32b6df07f#r155362349 ?